### PR TITLE
Fix for AccountClaim not registering Account CR being in an error state

### PR DESCRIPTION
**Pull request by ArielLima**
Moved the Account CR base case handler bellow AccountReady and AccountFailed tests, the base case was being checked too early causing the AccountFailed test never being reached. https://issues.redhat.com/browse/OSD-2627

### CODE REVIEW

**Comment by ArielLima** [comment]
I tested it locally and it works. please QC @fahlmant @jharrington22

File: pkg/controller/accountclaim/accountclaim_controller.go [hunk 1]
```diff
@@ -201,11 +201,26 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
 			return reconcile.Result{}, err
 		}
 
-		// Check account status requeue if we are not ready
 		if byocAccount.Status.State != string(awsv1alpha1.AccountReady) {
+			fmt.Println("ACCOUNT NOT READY")
```
Comment (fahlmant) [inline pkg/controller/accountclaim/accountclaim_controller.go]: Are these debug statements or meant to be shown in the logs? (Line 207 as well)

Comment (ArielLima) [reply pkg/controller/accountclaim/accountclaim_controller.go]: Debug, fixing now.

File: pkg/controller/accountclaim/accountclaim_controller.go [hunk 2]
```diff
@@ -223,11 +223,24 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
...
 		}
 
-		// Check account status requeue if we are not ready
 		if byocAccount.Status.State != string(awsv1alpha1.AccountReady) {
+			if byocAccount.Status.State == string(awsv1alpha1.AccountFailed) {
+				accountClaim.Status.State = awsv1alpha1.ClaimStatusError
+				message := "BYOC account errored"
+				accountClaim.Status.Conditions = controllerutils.SetAccountClaimCondition(
+					accountClaim.Status.Conditions,
+					awsv1alpha1.AccountClaimed,
```
Comment (fahlmant) [inline pkg/controller/accountclaim/accountclaim_controller.go]: Is this the right condition?

File: pkg/controller/accountclaim/accountclaim_controller.go [hunk 3]
```diff
@@ -223,11 +223,24 @@ func (r *ReconcileAccountClaim) Reconcile(request reconcile.Request) (reconcile.
...
+					corev1.ConditionTrue,
+					AccountClaimed,
```
Comment (fahlmant) [inline pkg/controller/accountclaim/accountclaim_controller.go]: Is this the right reason?

File: pkg/controller/accountclaim/accountclaim_controller.go (hunk 2 above)
Comment (ArielLima) [reply pkg/controller/accountclaim/accountclaim_controller.go]: agree, is this better?

File: pkg/controller/accountclaim/accountclaim_controller.go (hunk 3 above)
Comment (ArielLima) [reply pkg/controller/accountclaim/accountclaim_controller.go]: agree, is this better?

File: pkg/apis/aws/v1alpha1/accountclaim_types.go [hunk 4]
```diff
@@ -57,6 +57,8 @@ type AccountClaimConditionType string
 const (
 	// AccountClaimed is set when an Account is claimed
 	AccountClaimed AccountClaimConditionType = "Claimed"
+	// BYOCAccountClaimFailed is set when a BYOC Account Fails
+	BYOCAccountClaimFailed AccountClaimConditionType = "BYOCAccountFailed"
```
Comment (fahlmant) [inline pkg/apis/aws/v1alpha1/accountclaim_types.go]: I think this one should be `BYOCAccountClaimFailed` because its the state of the Account Claim itself

Comment (ArielLima) [reply pkg/apis/aws/v1alpha1/accountclaim_types.go]: done

**Comment by ArielLima** [comment]
Tested, working for me
cc @fahlmant

**Comment by rogbas** [comment]
@ArielLima can you please rebase it so we can test/approve it?

**Pull request merged**